IdentityCore/src/requests/sdk/adal/MSIDLegacyBrokerResponseHandler.m

```diff
@@ -44,7 +44,18 @@ @implementation MSIDLegacyBrokerResponseHandler
                                                   error:(NSError **)error
 {
 #if TARGET_OS_IPHONE
-    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:keychainGroup];
+    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:keychainGroup error:error];
+    
+    if (!dataSource)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Failed to initialize keychain cache.", nil, nil, nil, nil, nil);
```

Comment:
nit: should we add "make sure your app is properly signed and keychain access group is configured note..."?

Reply:
Sure updated the error message.